source/game.programme.newsevent.bmx

```diff
@@ -741,50 +781,59 @@ Type TNewsEvent Extends TBroadcastMaterialSource {_exposeToLua="selected"}
 	End Method
 
 
-	'ATTENTION:
-	'to emit an artificial news, use GetNewsAgency().announceNewsEvent()
-	Method doHappen(time:Long = 0)
-		if time = 0 then time = GetWorldTime().GetTimeGone()
-		
-		'set happened time, add to collection list...
-		GetNewsEventCollection().setNewsHappened(Self, time)
-
-		If time <= GetWorldTime().GetTimeGone()
-			'inform a template that it just happens
-			If templateID > 0
-				local newsTemplate:TNewsEventTemplate = GetNewsEventTemplateCollection().GetByID(templateID)
-				if newsTemplate
-					newsTemplate.OnHappen()
-				EndIf
-			Endif
+	'the news happened (now or earlier) and is now handled the first
+	'time
+	Method ProcessHappening(overrideHappenTime:Long = -1)
+		'override happened time?
+		if overrideHappenTime <> -1
+			self.happenedTime = overrideHappenTime
+		EndIf
 
-			'replace placeholders with CURRENT information (on creation
-			'of the news event the time could differ, or the weather
-			'is not so shiny...)
-			ReplacePlaceholdersFromTemplate(null, templateVariables, time)
+		If self.happenedTime > GetWorldTime().GetTimeGone()
+			Throw "TNewsEvent.ProcessHappening() - news event did not happen yet"
```

Comment:
Das ist das Sicherheitsnetz, dass Nachrichten erst verarbeitet werden, wenn sie wirklich passiert sind und damit Effekte zur richtigen Zeit wirksam werden. Richtig?

Reply:
So ungefaehr. Du kannst halt nun auch Newsevents "happen" lassen, die in der Vergangenheit liegen aber bisher halt noch nicht verarbeitet wurden